fullcalendar.js

```diff
@@ -0,0 +1,73 @@
+(function() {
+    'use strict';
+    const RADIOBUTTON = 'time';
+    const TEXTFIELD = 'event';
+    const DATEFIELD = 'date';
+    const STIME = 'start_time';
+    const ETIME = 'end_time';
+    const RADIOVALUE = 'Start time - End time';
+    const events = ['app.record.detail.show',
+                    'app.record.create.show',
+                    'app.record.create.change.' + RADIOBUTTON,
+                    'app.record.edit.show',
+                    'app.record.edit.change.' + RADIOBUTTON];
+
+    // Only show option to input start and end times if 'All day' not selected.
+    kintone.events.on(events, function(event) {
+        const record = event.record;
+        // Hide/unhide fields depending on field choices.
+        if (record[RADIOBUTTON].value === RADIOVALUE) {
+            kintone.app.record.setFieldShown(STIME, true);
+            kintone.app.record.setFieldShown(ETIME, true);
+        } else {
+            // If 'Start time - End time' is not selected, hide the Start time
+            // and End time fields.
+            kintone.app.record.setFieldShown(STIME, false);
+            kintone.app.record.setFieldShown(ETIME, false);
+        }
+    });
+
+    // Instantiate FullCalendar instance
+    kintone.events.on('app.record.index.show', function(event) {
+        if (event.viewType === 'custom') {
+            const calendarEl = document.getElementById('calendar');
+            const calendar = new FullCalendar.Calendar(calendarEl, {
+                plugins: ['dayGrid'],
+                displayEventEnd: true
+            });
+            const records = event.records;
```

Comment:
If you use `event.records`, you can only get records which are supposed to show up in the list view, which means that the maximum number of the records with the `event.records` is 100. Consider using `kintone.api()`&`/k/v1/records` so that you can fetch more records instead. It depends on what you want to achieve if you should change `event.records` to `kintone.api()`&`/k/v1/records`. It's ok If you are happy about the maximum number of the records.